src/parser/LiteralType.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023, University of Freiburg,
+//                 Chair of Algorithms and Data Structures.
+// Author: Benedikt Maria Beckermann <benedikt.beckermann@dagstuhl.de>
+
+#include "LiteralType.h"
+
+#include <utility>
+
+LiteralType::LiteralType(NormalizedString content) {
+  this->content = std::move(content);
+  this->descriptorType = LiteralDescriptor::NONE;
+}
+
+LiteralType::LiteralType(NormalizedString content,
+                         NormalizedString datatypeOrLanguageTag,
+                         LiteralDescriptor type) {
+  this->content = std::move(content);
+  this->descriptorType = type;
+  this->descriptorValue = std::move(datatypeOrLanguageTag);
+}
```

Comment:
Also use the direct initializers in the constructor (see above, and maybe also for your other classes.).